src/main/java/org/prebid/server/spring/config/server/application/ApplicationServerConfiguration.java

```diff
@@ -120,48 +154,35 @@ ExceptionHandler exceptionHandler(Metrics metrics) {
         return ExceptionHandler.create(metrics);
     }
 
-    @Bean("router")
-    Router router(BodyHandler bodyHandler,
-                  NoCacheHandler noCacheHandler,
-                  CorsHandler corsHandler,
-                  org.prebid.server.handler.openrtb2.AuctionHandler openrtbAuctionHandler,
-                  AmpHandler openrtbAmpHandler,
-                  VideoHandler openrtbVideoHandler,
-                  StatusHandler statusHandler,
-                  CookieSyncHandler cookieSyncHandler,
-                  SetuidHandler setuidHandler,
-                  GetuidsHandler getuidsHandler,
-                  VtrackHandler vtrackHandler,
-                  OptoutHandler optoutHandler,
-                  BidderParamHandler bidderParamHandler,
-                  BiddersHandler biddersHandler,
-                  BidderDetailsHandler bidderDetailsHandler,
-                  NotificationEventHandler notificationEventHandler,
-                  List<CustomizedAdminEndpoint> customizedAdminEndpoints,
-                  StaticHandler staticHandler) {
+    @Bean
+    BiFunction<String, SocketAddress, Verticle> applicationVerticleFactory(
+            @Qualifier("applicationServerRouter") Router router,
+            HttpServerOptions httpServerOptions,
+            ExceptionHandler exceptionHandler) {
+
+        return (name, address) -> new ServerVerticle(
+                name, httpServerOptions, address, router, exceptionHandler);
+    }
+
+    @Bean("applicationServerRouter")
+    Router applicationServerRouter(
+            BodyHandler bodyHandler,
+            NoCacheHandler noCacheHandler,
+            CorsHandler corsHandler,
+            List<ApplicationResource> resources,
+            @Qualifier("applicationPortAdminResourcesBinder") AdminResourcesBinder adminResourcesBinder,
+            StaticHandler staticHandler) {
```

Comment:
I don't see any other use of Vertx in this file, let's inject it directly to create this bean.